src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeBuffer.cs

```diff
@@ -248,7 +245,9 @@ public void ReadSpan<T>(ulong byteOffset, Span<T> buffer)
 
                 ref T structure = ref MemoryMarshal.GetReference(buffer);
                 for (int i = 0; i < buffer.Length; i++)
-                    Buffer.Memmove(ref Unsafe.Add(ref structure, i), ref Unsafe.AsRef<T>(ptr + alignedSizeofT * i), 1);
+                {
+                    Unsafe.Add(ref structure, (nint)(uint)i) = Unsafe.ReadUnaligned<T>(ptr + alignedSizeofT * (uint)i);
```

Comment:
This one its not clear why we aren't just using `span.CopyTo` or a single `Buffer.Memmove` instead of the for loop of individual `T`

Reply:
It would be a breaking change.

https://github.com/dotnet/runtime/blob/cb4c24e906f0d9fbe45123bdbe298edecf93d588/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeBuffer.cs#L385-L387

Reply:
Ah, I see, because its also factoring in the `aligned sizeof`, gotcha.